main/src/ca/uwaterloo/flix/tools/pkg/ManifestParser.scala

```diff
@@ -210,6 +214,26 @@ object ManifestParser {
     }
   }
 
+  /**
+    * Converts an optional String `optS` to an optional reference to a GitHub project.
+    * Returns an error if the string is present but not in the correct format.
+    * The only allowed format is "github:<username>/<repository>".
+    */
+  private def toGithubProject(optS: Option[String], p: Path): Result[Option[GitHub.Project], ManifestError] = {
```

Comment:
We should define this directly on a string and use the proper `map` or whatever function later

Reply:
Should we also introduce a `mapErr`?

Reply:
If it's needed, yes :)